src/docs/perf/rendering/shader.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Reduce shader compilation jank on mobile
+short-title: Shader jank
+description: What is shader jank and how to minimize it.
+---
+
+If the animations on your mobile app appear to be janky,
+but only on the first run, you can _warm up_ the
+shader for the Skia Shader Language (SkSL) for a
+significant improvement.
+
+![Side-by-side screenshots of janky mobile app next to non-janky app]({% asset perf/render/shader-jank.gif @path %}){:width="100%"}
+
+## What is shader compilation jank?
+
+If an app has janky animations during the first run,
+and later becomes smooth for the same animation,
+then it's very likely due to shader compilation jank.
+
+More technically, a shader is a piece of code that runs on
+a GPU (graphics processing unit).
+When a shader is first used, it needs to be compiled on the device.
+The compilation could cost up to a few hundred milliseconds
+whereas a smooth frame needs to be drawn within 16 milliseconds
+for a 60 fps (frame-per-second) display.
+Therefore, a compilation could cause tens of frames to be missed,
+and drop the fps from 60 to 6. This is _compilation jank_.
+After the compilation is complete, the animation should be smooth.
+
+Definitive evidence for the presence of shader compilation jank
+is to see `GrGLProgramBuilder::finalize` in the tracing.
+See the following screenshot for an example timeline tracing.
+
+![A tracing screenshot verifying jank]({% asset perf/render/tracing.png @path %}){:width="100%"}
+
+## What do we mean by "first run"?
+
+On Android, "first run" means that the user might see
+jank the first time opening the app. Subsequent opens
+should be fine.
+
+On iOS, "first run" means that the user might see
+jank the first time the user opens the app from scratch.
```

Comment:
`the first time the user opens the app from scratch` -> `when an animation first occurs every time the user opens the app from scratch`?

Reply:
Done